components/automate-cli/cmd/chef-automate/automateHAAwsDeployment.go

```diff
@@ -252,59 +272,173 @@ func (a *awsDeployment) validateEnvFields() *list.List {
 
 func (a *awsDeployment) validateCerts() *list.List {
 	errorList := list.New()
-	if a.config.Automate.Config.EnableCustomCerts {
-		if len(strings.TrimSpace(a.config.Automate.Config.PrivateKey)) < 1 ||
-			len(strings.TrimSpace(a.config.Automate.Config.PublicKey)) < 1 {
-			errorList.PushBack("Automate public_key and/or private_key are missing. Otherwise set enable_custom_certs to false.")
+
+	if checkIfFileExist(filepath.Join(initConfigHabA2HAPathFlag.a2haDirPath, "terraform", ".tf_arch")) {
+		err := a.getAwsHAIp()
+		if err != nil {
+			errorList.PushBack(fmt.Sprintf("error listing ips"))
 		}
-		// If root_ca is provided, check that it is valid
-		if len(strings.TrimSpace(a.config.Automate.Config.RootCA)) > 0 {
+	}
```

Comment:
```suggestion
	if checkIfFileExist(filepath.Join(initConfigHabA2HAPathFlag.a2haDirPath, "terraform", ".tf_arch")) {
		err := a.getAwsHAIp()
		if err != nil {
			return status.Wrap(err, status.IpAccessError, "Error in fetching node IPs in aws mode")
		}
	}
```
Move this to logic before validateConfigFields in generateConfig since this is not config validation